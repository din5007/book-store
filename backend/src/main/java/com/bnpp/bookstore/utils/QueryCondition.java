package com.bnpp.bookstore.utils;

import io.micrometer.common.util.StringUtils;
import org.springframework.stereotype.Component;

@Component("queryUtil")
public class QueryCondition {

  public boolean checkNull(String field) {
    return StringUtils.isBlank(field);
  }
}
