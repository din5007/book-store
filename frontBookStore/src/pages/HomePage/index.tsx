import { lazy } from 'react';

const HomePage = lazy(() => import('./HomePage'));

export default HomePage;