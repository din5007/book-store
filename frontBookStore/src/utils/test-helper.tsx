import { StoreProvider } from "easy-peasy";
import { render } from '@testing-library/react';
import { BrowserRouter } from "react-router-dom";


function customRender(component : any , store?: any) {

    return render(<StoreProvider store={store}>
        <BrowserRouter>
            {component}
        </BrowserRouter>
    </StoreProvider>);
}

export * from '@testing-library/react';
export { customRender } ;