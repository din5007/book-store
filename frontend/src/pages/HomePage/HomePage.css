.main {
    min-height: 93vh;
    display: flex;
    align-items: center;
  }
  
  .title {
    font-size: 95px;
  }
  
  .subtitle {
    font-size: 30px;
    padding-top: 10px;
  }
  
  .intro-text {
    width: 100%;
    text-align: center;
    padding-bottom: 80px;
  }
  
  .buttonContainer {
    margin-top: 60px;
    display: flex;
    justify-content: space-evenly;
  }
  
  .landingbutton {
    width: 200px;
    height: 55px;
  }
  
  @media only screen and (min-width: 1200px) {
    .buttonContainer {
      padding: 0 200px;
    }
    .title {
      padding: 0 100px;
    }
  }
  
  @media only screen and (min-width: 768px) {
    .landingbutton {
      width: 150px;
      height: 50px;
    }
    .title {
      font-size: 60px;
    }
  }