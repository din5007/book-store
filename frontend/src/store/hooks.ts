import { createTypedHooks } from "easy-peasy";

import { AppStoreModel } from ".";

export const { useStoreActions, useStoreState } = createTypedHooks<AppStoreModel>();