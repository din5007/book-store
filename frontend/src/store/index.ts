import { HttpService } from "../service/http/http.service";
import { createStore } from 'easy-peasy';
import { BookModel, booksModel } from "./books/books.model";
import { cartModel, CartModel } from "./cart/cart.model";
import { userModel, UserModel } from "./users/user.model";
import { OrderModel, orderModel } from "./order/order.model";

export interface AppStoreModel {
    books: BookModel;
    cart: CartModel;
    user: UserModel;
    order: OrderModel;
}

export const appStoreModel : AppStoreModel = {
    books: booksModel,
    cart: cartModel,
    user: userModel,
    order: orderModel,
}

export function storeModel() {
    return createStore(appStoreModel, {
        injections: {
            httpService: new HttpService()
        },
        initialState: {
            
        },
    })
}
export const store =  storeModel();