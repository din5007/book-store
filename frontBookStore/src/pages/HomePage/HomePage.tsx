import Container from "react-bootstrap/esm/Container";
import Button from "react-bootstrap/esm/Button";
import Row from "react-bootstrap/esm/Row";
import { useStoreState } from "../../store/hooks";

const HomePage = () => {
  const { isUserLoggedIn } = useStoreState(state => state.user);

  return (
    <div style={{
      minHeight: "93vh",
    display: "flex",
    alignItems: "center"
    }}>
      <Container>
        <Row>
          <div style={{
            width: "100%",
            textAlign: "center",
            paddingBottom: "80px"
          }}>
            <div>
              <h1 className="text-primary" style={{fontSize: "95px"}}>Welcome to Awesome Book Store</h1>
            </div>
            { !isUserLoggedIn && !localStorage.getItem('jwt') ? 
            <div style={{
              marginTop: "60px",
    display: "flex",
    justifyContent: "space-evenly"
            }}>
              <a href="/login">
                <Button size="lg" className="landingbutton" style={{width: "200px",
    height: "55px"}}>
                  Login
                </Button>
              </a>
              <a href="/register">
                <Button
                  size="lg"
                  className="landingbutton"
                  variant="outline-primary"
                >
                  SignUp
                </Button>
              </a>
            </div> : <></>}
          </div>
        </Row>
      </Container>
    </div>
  );
};

export default HomePage;