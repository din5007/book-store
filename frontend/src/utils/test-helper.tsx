import { StoreProvider } from "easy-peasy";
import { render } from '@testing-library/react';
import { AppStoreModel, store } from "../store";
import { Router } from 'react-router-dom';
import { ReactNode } from "react";



// const provider : React.FC<{children : ReactNode, store: any}> = ({ children, store }) => {
//     return (
        
//     )
// }

function customRender(component : any , store: any) {

    return render(<StoreProvider store={store}>
        {component}
    </StoreProvider>);
}

export * from '@testing-library/react';
export { customRender } ;