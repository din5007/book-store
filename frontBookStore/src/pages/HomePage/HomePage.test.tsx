import { createStore } from 'easy-peasy';
import { appStoreModel } from '../../store';
import { customRender as render } from '../../utils/test-helper';
import HomePage from "./HomePage"
import { describe, expect, it, vi } from 'vitest';

const mock = {
    get: vi.fn()
};

const store = createStore(appStoreModel, {
    injections: {
        httpService : mock
    },
    initialState: {
        
    }
})

describe('Home Page', () => {
    it('Home Page should load properly - snapshot', async () => {
        const {container}  = render(<HomePage />, store);
        expect(container).toMatchSnapshot();
    });
})