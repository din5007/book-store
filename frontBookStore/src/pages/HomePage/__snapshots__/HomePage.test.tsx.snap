// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home Page > Home Page should load properly - snapshot 1`] = `
<div>
  <div
    style="min-height: 93vh; display: flex; align-items: center;"
  >
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          style="width: 100%; text-align: center; padding-bottom: 80px;"
        >
          <div>
            <h1
              class="text-primary"
              style="font-size: 95px;"
            >
              Welcome to Awesome Book Store
            </h1>
          </div>
          <div
            style="margin-top: 60px; display: flex; justify-content: space-evenly;"
          >
            <a
              href="/login"
            >
              <button
                class="landingbutton btn btn-primary btn-lg"
                style="width: 200px; height: 55px;"
                type="button"
              >
                Login
              </button>
            </a>
            <a
              href="/register"
            >
              <button
                class="landingbutton btn btn-outline-primary btn-lg"
                type="button"
              >
                SignUp
              </button>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
